### PR TITLE
Social login with OpenShift: Metatype additions

### PR DESCRIPTION
**Pull request by ayoho**

### CODE REVIEW

**Comment by ayoho** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.oidcCertification,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.multiProvider

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -238,3 +238,19 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
 
 forwardLoginParameter=Forward login parameter
 forwardLoginParameter.desc=Specifies a comma-separated list of parameter names to forward to the OpenID Connect provider. If a protected resource request includes one or more of the specified parameters, the OpenID Connect client will include those parameters and their values in the authorization endpoint request to the OpenID Connect provider.
+
+k8sTokenReviewEndpoint=Kubernetes Token Review Endpoint
+k8sTokenReviewEndpoint.desc=URL for the Kubernetes TokenReview API that is used to obtain information about an authenticated user from an access token.
+
+serviceAccountTokenForK8sTokenreview=Service account token for Kubernetes Token Review
+serviceAccountTokenForK8sTokenreview.desc=Token for OpenShift service account with the system:auth-delegator role. The token is required for making requests to the Kubernetes TokenReview API.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change
"The token is required for making requests to the Kubernetes TokenReview API."
To
"**This** token is required for making requests to the Kubernetes TokenReview API."

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -238,3 +238,19 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
...
+
+useAccessTokenFromRequest=Use access token from request
+useAccessTokenFromRequest.desc=Determines if the access token that is provided in the request is used for authentication.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change
"Determines if the access token that is provided in the request is used for authentication."
To
"Determines **whether** the access token that is provided in the request is used for authentication."

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -238,3 +238,19 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
...
+
+useAccessTokenFromRequest.no=Do not use the access token that is provided in the request for authentication.
+useAccessTokenFromRequest.required=The access token that is provided in the request must be used for authentication.
+useAccessTokenFromRequest.ifPresent=Use the access token that is provided in the request for authentication if it is present.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change
"Use the access token that is provided in the request for authentication if it is present."
To
"If an access token is provided, use it in the request for authentication. "

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -238,3 +238,19 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
...
+
+tokenHeaderName=OAuth token header name
+tokenHeaderName.desc=Name of the header to use when forwarding an OAuth token.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change
"Name of the header to use when forwarding an OAuth token."
To
Name of the header to use when an OAuth token is forwarded."

**Comment by ayoho** [comment]
Had to push a new update based on feature design changes. But was still able to incorporate a couple suggestions from the ID review.

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -238,3 +238,19 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
...
+
+# Do not translate "OpenShift", "TokenReview", "system:auth-delegator"
+userApiToken=Access token for user API calls
+userApiToken.desc=Specifies the access token that has permission to call the user API. For OpenShift, the token is from a service account with permission to call the TokenReview API, which usually requires the system:auth-delegator role.
+
+accessTokenAuthentication=Use access token authentication
+accessTokenAuthentication.desc=Determines whether the access token that is provided in the request is used for authentication. If true, the client must provide a valid access token.
+
+accessTokenAuthenticationSupported=Access token authentication supported
+accessTokenAuthenticationSupported.desc=Determines whether to support access token authentication if an access token is provided in the request. If true, and an access token is provided in the request, the access token will be used as an authentication token.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: @ayoho Thanks for making the updates. I see you pushed some more changes, and I just have one small suggested update. Let me know if you have any questions. 

Change
"If true, and an access token is provided in the request, the access token will be used as an authentication token."
To
"If true, and an access token is provided in the request, the access token **is** used as an authentication token."

**Comment by ayoho** [comment]
Pushed new changes yet again to account for feature design changes and attribute renames.

#build

**Comment by c00crane** [comment]
Did you run the Social FATs (full mode) against the changes?  (just to make sure nothing regressed)

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 6]
```diff
@@ -238,3 +238,23 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
 
 forwardLoginParameter=Forward login parameter
 forwardLoginParameter.desc=Specifies a comma-separated list of parameter names to forward to the OpenID Connect provider. If a protected resource request includes one or more of the specified parameters, the OpenID Connect client will include those parameters and their values in the authorization endpoint request to the OpenID Connect provider.
+
+userApiType=User API specification
+userApiType.desc=Indicates the specification to use for the user API.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change:
"Indicates the specification to use for the user API."
To
"Indicates **which** specification to use for the user API."

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 7]
```diff
@@ -238,3 +238,23 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
...
+
+userApiType.basic=Indicates that the user API can be directly invoked with an HTTP GET request.
+# Do not translate "Kubernetes", "TokenReview"
+userApiType.kube=Indicates that the user API is a Kubernetes TokenReview API
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Add a period at the end of "Indicates that the user API can be directly started with an HTTP GET request".

File: dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties [hunk 8]
```diff
@@ -238,3 +238,23 @@ discoveryPollingRate.desc=Rate in milliseconds at which the OpenID Connect clien
 
 forwardLoginParameter=Forward login parameter
 forwardLoginParameter.desc=Specifies a comma-separated list of parameter names to forward to the OpenID Connect provider. If a protected resource request includes one or more of the specified parameters, the OpenID Connect client will include those parameters and their values in the authorization endpoint request to the OpenID Connect provider.
+
+userApiType=User API specification
+userApiType.desc=Indicates the specification to use for the user API.
+
+userApiType.basic=Indicates that the user API can be directly invoked with an HTTP GET request.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/OSGI-INF/l10n/metatype.properties]: Change
"Indicates that the user API can be directly invoked with an HTTP GET request."
To
"Indicates that the user API can be directly **started** with an HTTP GET request."

**Comment by ayoho** [comment]
Pushed final minor updates based on ID review. Will also wait on the results of new personal build https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_QzkhwAFfEeqlbYcQx7mKkA before delivering to ensure existing social FATs do not break.

**Pull request merged**